### PR TITLE
parser: Fix CHECK PARTITION syntax

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 
to #435 . Support `CHECK PARTITION` Syntax. 

### What is changed and how it works?
1. Add related parser rule in `parser.y`
2. Add a helper struct `AllOrPartitionNames` and complete `Restore` method for struct `AlterTableSpec`
3. Add related tests

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

N / A

Side effects

N / A

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -702,6 +702,7 @@ import (
 
 %type   <item>
 	AdminShowSlow			"Admin Show Slow statement"
+	AllOrPartitionNameList	"All or partition list"
```
[the remark this answers is not in the document]
Comment (spongedu) [reply parser.y]: Done. @tangenta  PTAL :)

**Comment by tangenta** [comment]
@spongedu Due to our negligence, it seems there are some conflicts with [the PR](https://github.com/pingcap/parser/pull/453/files). You can pause updating until some reusable data structures are developed.

**Comment by spongedu** [comment]
> @spongedu Due to our negligence, it seems there are some conflicts with [the PR](https://github.com/pingcap/parser/pull/453/files). You can pause updating until some reusable data structures are developed.

OK.

**Comment by spongedu** [comment]
@leoppro @tangenta PTAL

File: ast/ddl.go [hunk 2]
```diff
@@ -1882,6 +1883,11 @@ func (a AlgorithmType) String() string {
 	}
 }
 
+type AllOrPartitionNames struct {
+	All            bool
+	PartitionNames []model.CIStr
+}
+
```
Comment (tangenta) [inline ast/ddl.go]: Do we need this structure anymore?

Comment (spongedu) [reply ast/ddl.go]: > Do we need this structure anymore?

No, it's useless now. I'll remove it.

**Pull request merged**